gerber/layers.py

```diff
@@ -140,11 +140,18 @@ def guess_layer_class(filename):
                 if re.findall(hint.regex, filename, re.IGNORECASE):
                     return hint.layer
 
-            patterns = [r'^(\w*[.-])*{}([.-]\w*)?$'.format(x) for x in hint.name]
-            if ext[1:] in hint.ext or any(re.findall(p, name, re.IGNORECASE) for p in patterns):
-                return hint.layer
+            if ext[1:] == 'gbr':									#Prateek
+                patterns = [r'(\w*[.-])*{}([.-]\w*)?$'.format(x) for x in hint.name]			#Prateek
+                if any(re.findall(p, name, re.IGNORECASE) for p in patterns):				#Prateek
+                    return hint.layer									#Prateek
+            else :											#Prateek
+                patterns = [r'^(\w*[.-])*{}([.-]\w*)?$'.format(x) for x in hint.name]
+                if ext[1:] in hint.ext or any(re.findall(p, name, re.IGNORECASE) for p in patterns):
+                    return hint.layer
+            

```

Comment:
None of these changes are necessary if you remove 'gbr' from the extension hints above

Reply:
Yes if I remove 'gbr' from the extension hints these changes are not required, but one change is required and that is in defining patterns :
patterns = [r'(\w*[.-])*{}([.-]\w*)?$'.format(x) for x in hint.name]
instead of 
patterns = [r'^(\w*[.-])*{}([.-]\w*)?$'.format(x) for x in hint.name]
according to my changes to name attribute in hints.

Reply:
there is * before {} in defining patterns, i.e.;
patterns = [r'(\w*[.-])*{}([.-]\w*)?$'.format(x) for x in hint.name]